recipe/meta.yaml

```diff
@@ -9,11 +9,16 @@ source:
   sha256: 5db25d4fd138013b563f9a3d1d87f7de7df1dac014fd4cccdfbb3435a5cff761
 
 build:
-  number: 1
-  skip: true  # [osx or win]
+  number: 2
+  noarch: generic
 
 requirements:
   run:
+    - __linux               # [linux]
+    - __win                 # [win]
+    - __archspec * x86_64   # [x86_64]
```

Comment:
Interesting this use to be a valid way to check `x86_64` generally

Though think some things in the stack have shifted due to recent work around `archspec`. As a result this now errors `__archspec` reports the microarchitecture. For example on CI we can see this is `broadwell` (see below)

So we may need another approach

<details>

<summary>Result of <code>conda info</code>:</summary>

```
     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /home/conda/.condarc
 populated config files : /opt/conda/.condarc
                          /home/conda/.condarc
          conda version : 23.7.4
    conda-build version : 3.24.0
         python version : 3.10.13.final.0
       virtual packages : __archspec=1=broadwell
                          __glibc=2.17=0
                          __linux=6.2.0=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/conda/feedstock_root/build_artifacts/pkg_cache
                          /opt/conda/pkgs
       envs directories : /opt/conda/envs
                          /home/conda/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.4 requests/2.31.0 CPython/3.10.13 Linux/6.2.0-1015-azure centos/7.9.2009 glibc/2.17
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False
```

</details>

Reply:
@conda-forge/core , is there another approach we should consider taking here in light of the `archspec` changes?

Reply:
i feel like `__linux` should be expanded to
```
__linux_64
__linux_ppc64le
__linux_aarch64
```

similarly for osx.

I did not know a way to better differentiate for noarch packages, so I omitted a noarch optimization due to similar constraint

Reply:
Opened issue ( https://github.com/conda/conda/issues/13420 ) to discuss how to capture macroarchitecture information

Reply:
Why would you need this package to be `noarch: generic` in the first place? If you want to check `linux` and `x86_64`, simply don't make it noarch.